### ISSUE TITLE
Unexpected NULL values in the preprocessed .csv file

### ISSUE THREAD

**Issue by SoccerCashUK** [MauroAndretta/AIFootballPredictions#5]
Hi get this 

----------------------------------

Making predictions for E0...

Loaded model and data for E0.
Predicting matches for E0...
Predictions made for E0.
----------------------------------

Making predictions for I1...

Loaded model and data for I1.
Predicting matches for I1...
Predictions made for I1.
----------------------------------

Making predictions for D1...

Loaded model and data for D1.
Predicting matches for D1...
Predictions made for D1.
----------------------------------

Making predictions for SP1...

Loaded model and data for SP1.
Predicting matches for SP1...
Traceback (most recent call last):
  File "C:\Users\Iphon\Downloads\AIFootballPreictions-main\AIFootballPreictions-main\scripts\make_predictions.py", line 209, in <module>
    main(args.input_leagues_models_dir, args.input_data_predict_dir, args.final_predictions_out_file, args.json_competitions)
  File "C:\Users\Iphon\Downloads\AIFootballPreictions-main\AIFootballPreictions-main\scripts\make_predictions.py", line 192, in main
    league_section = make_predictions(league, league_model, league_data, competitions)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\Downloads\AIFootballPreictions-main\AIFootballPreictions-main\scripts\make_predictions.py", line 147, in make_predictions
    prediction = league_model.predict(X_test)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\ensemble\_voting.py", line 440, in predict
    maj = np.argmax(self.predict_proba(X), axis=1)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\ensemble\_voting.py", line 481, in predict_proba
    self._collect_probas(X), axis=0, weights=self._weights_not_none
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\ensemble\_voting.py", line 456, in _collect_probas
    return np.asarray([clf.predict_proba(X) for clf in self.estimators_])
                       ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\linear_model\_logistic.py", line 1431, in predict_proba
    return super()._predict_proba_lr(X)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\linear_model\_base.py", line 397, in _predict_proba_lr
    prob = self.decision_function(X)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\linear_model\_base.py", line 363, in decision_function
    X = self._validate_data(X, accept_sparse="csr", reset=False)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\base.py", line 633, in _validate_data
    out = check_array(X, input_name="X", **check_params)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\utils\validation.py", line 1064, in check_array
    _assert_all_finite(
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\utils\validation.py", line 123, in _assert_all_finite
    _assert_all_finite_element_wise(
  File "C:\Users\Iphon\AppData\Local\Programs\Python\Python312\Lib\site-packages\sklearn\utils\validation.py", line 172, in _assert_all_finite_element_wise
    raise ValueError(msg_err)
ValueError: Input X contains NaN.
LogisticRegression does not accept missing values encoded as NaN natively. For supervised learning, you might want to consider sklearn.ensemble.HistGradientBoostingClassifier and Regressor which accept missing values encoded as NaNs natively. Alternatively, it is possible to preprocess the data, for instance by using an imputer transformer in a pipeline or drop samples with missing values. See https://scikit-learn.org/stable/modules/impute.html You can find a list of all estimators that handle NaN values at the following page: https://scikit-learn.org/stable/modules/impute.html#estimators-that-handle-nan-values

**Comment by MauroAndretta** [comment]
This is strange!

It seems like there are Null values in the .csv file related to the Spanish league. 

As a workaround, for the moment you can try to delete the `.csv `file related to the Spanish League (the one with SP1 in the name) in the folder `data/preprocessed` as well as delete the model related to the Spanish League  (the one with SP1 in the name) in the folder `models`.

let me know if this solve the issue. 


Sorry for that it was not expected!
I'll try some tests on my side, eventually updating the preprocessing script.

**Comment by SoccerCashUK** [comment]
Hi now its doing to F1

James

**Comment by MauroAndretta** [comment]
Is there the same error with the F1?

Really sorry, please do the same also F1 related csv and model.

I'll try to solve this issue as soon as possible

**Comment by SoccerCashUK** [comment]
Hi i did that and it works now, no need to be sorry we get there in the end 
please let me know once all fixed.
I also left you a star

James

**Comment by MauroAndretta** [comment]
There is something I have to fix during the handling of missing values. 

I will put a comment here to update you!

Thanks James, bye! ☺️

**Comment by MauroAndretta** [comment]
With the last push, I have updated the `make_predictions.py` script to resolve the issue.

Issue: After the feature selection step, some selected features were not present in the GENERAL_FEATURE list within the `make_predictions.py` script.

**Fix**: I addressed this by considering all features that do not belong to the home or away team as general features. This ensures all selected features are appropriately categorized and available for use.

A simple git pull from the main branch (to update your local code with the latest changes) will resolve the issue. This issue can now be closed.
